File: code/backend/server_py/src/module_rag/controller/user_model.py
from fastapi import APIRouter, HTTPException, status, Depends
from module_rag.do.user_model import UserModelUpdate, UserModelResponse
from module_rag.service.user_model import UserModelService
from module_rag.dependencies.user_model import get_user_model_service
from module_authorization.dependencies.auth import get_current_user_id
from module_rag.config.server import module_app

router = APIRouter()


@router.get(
    "/my", summary="获取当前用户的模型绑定", response_model=UserModelResponse
)
async def get_my_model_binding(
    current_user_id: str = Depends(get_current_user_id),
    service: UserModelService = Depends(get_user_model_service),
):
    """
    获取当前登录用户的模型绑定(chat模型/向量化模型)
    :param current_user_id: 当前登录用户ID(由 token 自动解析)
    :param service: 用户-模型绑定服务依赖注入
    :return: 用户-模型绑定详情
    """
    try:
        result = await service.get_by_user(current_user_id)
        if not result:
            # 未绑定则返回空绑定
            return UserModelResponse(
                id="",
                user_id=current_user_id,
                chat_model_id=None,
                embedding_model_id=None,
                created_at=None,
                updated_at=None,
            )
        return result
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e)
        )


@router.put(
    "/my", summary="更新当前用户的模型绑定", response_model=UserModelResponse
)
async def update_my_model_binding(
    user_model: UserModelUpdate,
    current_user_id: str = Depends(get_current_user_id),
    service: UserModelService = Depends(get_user_model_service),
):
    """
    更新当前登录用户的模型绑定(chat模型/向量化模型)
    :param user_model: 更新数据(chat_model_id/embedding_model_id)
    :param current_user_id: 当前登录用户ID(由 token 自动解析)
    :param service: 用户-模型绑定服务依赖注入
    :return: 更新后的用户-模型绑定详情
    """
    try:
        return await service.upsert(current_user_id, user_model)
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e)
        )


# 注册路由
module_app.include_router(router, prefix="/user-models", tags=["用户模型绑定"])

File: code/frontend/vue3/src/modules/rag/api/user_model.ts
// src/modules/rag/api/user_model.ts
// 用户-模型绑定 API(对应后端 /rag/user-models/*, v4 4.3)
import { http_base_server } from '@/common/api/http';

/** 当前用户模型绑定(未绑定字段为 null) */
export interface UserModelBinding {
  id: string;
  user_id: string;
  /** 绑定的对话模型配置ID */
  chat_model_id: string | null;
  /** 绑定的向量化模型配置ID */
  embedding_model_id: string | null;
  /** 绑定的重排模型配置ID */
  rerank_model_id: string | null;
  /** 回退开关: true=绑定失效时不回退默认公共模型(直接报错) */
  fallback_disabled: boolean;
  created_at: string | null;
  updated_at: string | null;
}

/** 更新模型绑定载荷(仅提交的字段生效; 显式 null 表示解绑) */
export interface UserModelBindingUpdate {
  chat_model_id?: string | null;
  embedding_model_id?: string | null;
  rerank_model_id?: string | null;
  fallback_disabled?: boolean;
}

/**
 * 获取当前用户的模型绑定(未绑定时返回空绑定结构)
 * @returns 用户模型绑定详情
 */
export const getMyModelBinding = () => {
  return http_base_server.get<UserModelBinding>('/rag/user-models/my');
};

/**
 * 更新当前用户的模型绑定(绑定前校验模型归属: 公共/本部门/本人模型)
 * @param payload 更新载荷(未提交字段保持原值; 模型ID传 null 解绑)
 * @returns 更新后的绑定详情
 */
export const updateMyModelBinding = (payload: UserModelBindingUpdate) => {
  return http_base_server.put<UserModelBinding>('/rag/user-models/my', payload);
};
